tests/test_run.py

```diff
@@ -90,3 +108,57 @@ def test_main_match(test_directory):
 
     with open(mock_expected_path) as expected:
         assert result_file.read() == expected.read()
+
+
+def test_scraper(monkeypatch, test_directory):
+    def mock_get_mapping(_):
+        return [
+            {'property': 'P321', 'url': 'http://schema.org/director'},
+            {'property': 'P123', 'url': 'http://schema.org/name'},
+        ]
+
+    def mock_normalize_from_extruct(_, data):
+        return [{
+            "http://schema.org/director": [
+                data['microdata'][0]['director']['name']
+            ],
+            "http://schema.org/genre": [
+                data['microdata'][0]['genre']
+            ]
+        }]
+
+    monkeypatch.setattr(SchemaOrgNormalizer, "normalize_from_extruct", mock_normalize_from_extruct)
+    monkeypatch.setattr(WdqsSchemaorgPropertyMapper, "get_mapping", mock_get_mapping)
+    monkeypatch.setattr(requests, "Session", MockSession)
+
+    test_given_filename = "test_given_pipe1.jsonl"
+    mock_input_path = relative_path('mock_data', test_given_filename)
+    given_file = test_directory.join('data', test_given_filename)
+    shutil.copy(mock_input_path, given_file.strpath)
+
+    side_service_file = test_directory.join('data', 'side_service_input.json')
+    shutil.copy(relative_path('mock_data', 'empty.json'), side_service_file.strpath)
+
+    test_filename = "test_result_scraper.jsonl"
+    mock_file_path = test_directory.join('scripts', 'this_is_ignored.py')
+    result_file = test_directory.join('data', test_filename)
+
+    mock_args = f"this_is_ignored.py --step scrape " \
+                f"--input {test_given_filename} --output {test_filename}"
+
+    run_main(mock_args.split(), mock_file_path)
+
+    expected_result = {
+        'statement': {
+            'pid': 'P321',
+            'datatype': 'wikibase-item',
+            'value': {'numeric-id': 214917, 'id': 'Q214917'}},
+        'itemId': 'Q42',
+        'reference': {'referenceMetadata': {
+            'a': 'b',
+            'P854': 'https://example_with_schema.org/wow'},
+            'extractedData': ['James Cameron']}}
```

Comment:
Silly nitpick: Can you expand the dictionary here in a clearer way, it kinda looks like `a` and `P854` are children of `reference` rather than `referenceMetadata`.

Reply:
oh sure. it looks ewww right now.

Reply:
Done!